### PR TITLE
Launch etcd in coordinator

### LINKED ISSUE #875: [BUG] etcd should be launched by coordiantor
**Describe the bug**

then pass the endpoint to vineyard, and don't assume the default port "localhost:2379" for zetcd under local session as well.

Also, clear the etcd data when closing the session.

**Additional context**

This task should be done before delivering the next wheel release to our end-users.

### LINKED ISSUE #876: refine the function `get_free_port` and make it widely used.
1. refs: https://www.programcreek.com/python/?CodeExample=get+free+port
2. then revisit these code, use this function instead. 
https://github.com/alibaba/GraphScope/search?p=1&q=random+port

related issue: #875

### PR DESCRIPTION
**Pull request by lidongze0629**
## Related issue number

Fixes #875 
Fixes #876

### CODE REVIEW

**Review by sighingnow — APPROVED** [review]
Add `etcd-distro` to the dependency of `graphscope` (not `graphscope-client`).

File: coordinator/gscoordinator/launcher.py [hunk 1]
```diff
@@ -263,18 +266,77 @@ def close_interactive_instance(self, object_id):
         )
         return process
 
+    def _launch_etcd(self):
+        self._etcd_listen_peer_port = get_free_port("localhost")
```
Comment (yecol) [inline coordinator/gscoordinator/launcher.py:270]: We'd better try to use the default port, 2379/2380, only get a new if occupied.

File: coordinator/gscoordinator/launcher.py [hunk 2]
```diff
@@ -263,18 +266,77 @@ def close_interactive_instance(self, object_id):
...
+        self._etcd_listen_client_port = get_free_port("localhost")
+
+        etcd_exec = shutil.which("etcd")
```
Comment (yecol) [inline coordinator/gscoordinator/launcher.py:273]: move to line-270, check first.

File: coordinator/gscoordinator/launcher.py (hunk 1 above)
Comment (yecol) [reply coordinator/gscoordinator/launcher.py:270]: name it _etcd_peer_port/_etcd_client_port may be more concise.

File: coordinator/gscoordinator/launcher.py [hunk 3]
```diff
@@ -263,18 +266,77 @@ def close_interactive_instance(self, object_id):
...
+                raise RuntimeError("Launch etcd service failed.")
+        logger.info(
+            "Etcd is ready, endpoint is localhost:{0}".format(
+                self._etcd_listen_client_port
+            )
+        )
+
     def _launch_zetcd(self):
-        self._zookeeper_port = self._get_free_port(self._hosts.split(",")[0])
+        self._zookeeper_port = get_free_port(self._hosts.split(",")[0])
```
Comment (yecol) [inline coordinator/gscoordinator/launcher.py:329]: ditto, default port is preferred.

```
port = is_free_port(DEFAULT_PORT)? DEFAULT_PORT: get_free_port()
```

File: python/graphscope/framework/utils.py [hunk 4]
```diff
@@ -33,6 +34,44 @@
 from graphscope.proto import types_pb2
 
 
+def is_free_port(port, host="localhost", timeout=1.0):
```
Comment (yecol) [inline python/graphscope/framework/utils.py:37]: 1 second? Please double check whether this empirical value is reasonable for a port testing.

**Review by yecol — CHANGES_REQUESTED** [review]

File: coordinator/gscoordinator/launcher.py [hunk 5]
```diff
@@ -263,18 +266,77 @@ def close_interactive_instance(self, object_id):
...
+
+        start_time = time.time()
+
+        while is_free_port(self._etcd_listen_client_port):
+            time.sleep(1)
+            if (
+                self._timeout_seconds
+                and self._timeout_seconds + start_time < time.time()
+            ):
+                raise RuntimeError("Launch etcd service failed.")
```
Comment (yecol) [inline coordinator/gscoordinator/launcher.py:321]: give detailed error message with reason. e.g., 
`Launch ectd failed due to timeout.`

File: python/graphscope/framework/utils.py (hunk 4 above)
Comment (lidongze0629) [reply python/graphscope/framework/utils.py:37]: following [mysql](https://www.programcreek.com/python/?code=mysql%2Fmysql-utilities%2Fmysql-utilities-master%2Fmysql-test%2Fmutlib%2Fmutlib.py), change timeout to 0.2

**Review by yecol — APPROVED** [review]

**Pull request merged**